src/io/ipc/read/reader.rs

```diff
@@ -236,7 +236,7 @@ impl<R: Read + Seek> FileReader<R> {
     /// Panics iff the projection is not in increasing order (e.g. `[1, 0]` nor `[0, 1, 1]` are valid)
     pub fn new(reader: R, metadata: FileMetadata, projection: Option<Vec<usize>>) -> Self {
         if let Some(projection) = projection.as_ref() {
-            let _ = projection.windows(2).for_each(|x| {
+            projection.windows(2).for_each(|x| {
```

Comment:
Looks wonderful! Maybe we can add a test for this to make sure it doesn't panic now?

Reply:
Pointer for the tests: https://github.com/jorgecarleitao/arrow2/blob/main/tests/it/io/ipc/read/file.rs#L181 (changing from 1 to 0 should be enough :) )